src/devices/bus/psx/parallel.cpp

```diff
@@ -0,0 +1,119 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+/**********************************************************************
+
+        Sony Playstation 'Parallel' slot
+
+**********************************************************************/
+
+#include "emu.h"
+#include "parallel.h"
+
+//**************************************************************************
+//  DEVICE DEFINITIONS
+//**************************************************************************
+
+DEFINE_DEVICE_TYPE(PSX_PARALLEL_SLOT, psx_parallel_slot_device, "psx_parallel_slot", "Playstation Parallel Slot")
+
+
+//**************************************************************************
+//  DEVICE SPECTRUM_EXPANSION CARD INTERFACE
+//**************************************************************************
+
+//-------------------------------------------------
+//  psx_parallel_interface - constructor
+//-------------------------------------------------
+
+psx_parallel_interface::psx_parallel_interface(const machine_config &mconfig, device_t &device)
+	: device_slot_card_interface(mconfig, device)
+{
+	m_slot = dynamic_cast<psx_parallel_slot_device *>(device.owner());
+}
+
+
+//-------------------------------------------------
+//  ~psx_parallel_interface - destructor
+//-------------------------------------------------
+
+psx_parallel_interface::~psx_parallel_interface()
+{
+}
+
+//**************************************************************************
+//  LIVE DEVICE
+//**************************************************************************
+
+//-------------------------------------------------
+//  psx_parallel_slot_device - constructor
+//-------------------------------------------------
+
+psx_parallel_slot_device::psx_parallel_slot_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, PSX_PARALLEL_SLOT, tag, owner, clock),
+	device_slot_interface(mconfig, *this),
+	m_card(nullptr)
+{
+}
+
+//-------------------------------------------------
+//  expansion_slot_device - destructor
+//-------------------------------------------------
+
+psx_parallel_slot_device::~psx_parallel_slot_device()
+{
+}
+
+//-------------------------------------------------
+//  device_start - device-specific startup
+//-------------------------------------------------
+
+void psx_parallel_slot_device::device_start()
+{
+	m_card = dynamic_cast<psx_parallel_interface *>(get_card_device());
+}
+
+//-------------------------------------------------
+//  device_reset - device-specific reset
+//-------------------------------------------------
+
+void psx_parallel_slot_device::device_reset()
+{
+	if (get_card_device())
+	{
+		get_card_device()->reset();
```

Comment:
The `device_t` implementation of `reset()` resets child devices automatically.  You don't need to do this.

Reply:
not my code, copied from the spectrum port stuff, please fix it there if you don't want people copy+pasting it.

Reply:
I don't want people copy/pasting anything if they don't understand it, and if I knew it was being done there as well, I would have commented on the commit and asked the author to fix it.  And anything you copy/paste is your code, and you have no excuse for bringing problematic baggage along with it.

Reply:
Also, do you seriously think I have time to go through all of MAME and fix everything in case someone copy/pastes it?  You seriously over-estimate the amount of spare time I have.